components/style/properties/gecko.mako.rs

```diff
@@ -1033,23 +1033,48 @@ fn static_assert() {
         self.gecko.mImage.mPositionYCount = cmp::min(1, other.gecko.mImage.mPositionYCount);
         self.gecko.mImage.mLayers.mFirstElement.mPosition =
             other.gecko.mImage.mLayers.mFirstElement.mPosition;
+        unsafe {
+            Gecko_EnsureImageLayersLength(&mut self.gecko.mImage, other.gecko.mImage.mLayers.len());
+        }
+        for (layer, other) in self.gecko.mImage.mLayers.iter_mut()
```

Comment:
Not in this case, because the count is stored separately.

While we have a single vector per longhand (SoA), Gecko has one mega-vector (AoS) with separate length parameters for each longhand. This is specific to nsStyleImageLayers (the `background` struct), so doesn't really make sense on nsArray